src/main/java/com/google/lecturechat/data/DatastoreAccess.java

```
@@ -202,4 +203,101 @@ public void addEventToGroup(
     }
     return events;
   }
+
+  /**
+   * Adds the user to the database if they don't exist already.
+   *
+   * @param userId The id of the user that will be added.
+   * @param name The name of the user that will be added.
+   */
+  public void addUser(String userId, String name) {
+    Transaction transaction = datastore.beginTransaction();
+
+    try {
+      if (getUser(userId) == null) {
+        Entity userEntity = new Entity(KeyFactory.createKey(UserEntity.KIND.getLabel(), userId));
+        userEntity.setProperty(UserEntity.NAME_PROPERTY.getLabel(), name);
+        userEntity.setProperty(UserEntity.GROUPS_PROPERTY.getLabel(), new ArrayList<Long>());
+        datastore.put(userEntity);
+      }
+      transaction.commit();
+    } finally {
+      if (transaction.isActive()) {
+        transaction.rollback();
+      }
+    }
+  }
+
+  /**
+   * Searches for the user with the given id and returns the entity associated (or null if there is
+   * no user associated with the id received).
+   *
+   * @param userId The id of the user to be searched for.
+   * @return The entity associated or null if the user is not registered.
+   */
+  public Entity getUser(String userId) {
+    Query query = new Query(KeyFactory.createKey(UserEntity.KIND.getLabel(), userId));
+    Entity userEntity = datastore.prepare(query).asSingleEntity();
```

Seems like there is a simpler way of getting the entry with the specified key, by using datastore.get();
Also, why don't you use the method getEntityById implemented further above which does exactly the same? (except for throwing an exception instead of returning an empty Optional.